File: testdata/bench/javascript/BenchmarkTest00008.js
// Safe: SQL injection prevented via Sequelize replacements
const express = require('express');
const sequelize = require('./sequelize');
const router = express.Router();
router.get('/test008', (req, res) => {
    const param = req.query.username;
    sequelize.query("SELECT * FROM users WHERE username = :name", { replacements: { name: param } })
        .then(results => res.json(results))
        .catch(err => res.status(500).json({ error: 'Query failed' }));
});
module.exports = router;

File: testdata/bench/rust/BenchmarkTest00008.rs
use actix_web::{web, HttpResponse};
use std::collections::HashMap;

// Safe: SQL injection prevented with rusqlite params
async fn handler(query: web::Query<HashMap<String, String>>) -> HttpResponse {
    let param = query.get("id").unwrap_or(&String::new()).clone();
    // conn.execute("SELECT * FROM users WHERE id = ?1", rusqlite::params![&param])
    HttpResponse::Ok().body("safe")
}
